diff --git a/deps/gleq/gleq.h b/deps/gleq/gleq.h
index 1e02f41..7ffa53a 100644
--- a/deps/gleq/gleq.h
+++ b/deps/gleq/gleq.h
@@ -67,7 +67,7 @@ typedef enum
 
 } GLEQtype;
 
-typedef struct
+typedef struct GLEQevent
 {
     GLEQtype type;
     GLFWwindow* window;
@@ -271,7 +271,7 @@ static void gleqFileDropCallback(GLFWwindow* window, int count, const char** pat
     GLEQevent* event = gleqNewEvent();
     event->type = GLEQ_FILE_DROPPED;
     event->window = window;
-    event->file.paths = malloc(count * sizeof(char*));
+    event->file.paths = (char**)malloc(count * sizeof(char*));
     event->file.count = count;
 
     while (count--)
@@ -292,8 +292,10 @@ void gleqTrackWindow(GLFWwindow* window)
     glfwSetCursorEnterCallback(window, gleqCursorEnterCallback);
     glfwSetScrollCallback(window, gleqScrollCallback);
     glfwSetKeyCallback(window, gleqKeyCallback);
+#ifndef EMSC
     glfwSetCharModsCallback(window, gleqCharModsCallback);
     glfwSetDropCallback(window, gleqFileDropCallback);
+#endif
 }
 
 int gleqNextEvent(GLEQevent* event)
